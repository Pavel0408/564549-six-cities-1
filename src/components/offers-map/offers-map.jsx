// import leaflet from "leaflet";
import PropTypes from "prop-types";
import React, {PureComponent} from "react";

export class OffersMap extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    return <div id="map" style={{height: 823}}></div>;
  }

  componentDidMount() {
    const {offers} = this.props;
    this.props.mapService.render(offers);
  }
}

OffersMap.propTypes = {
  offers: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    rating: PropTypes.number.isRequired,
    isPremium: PropTypes.bool.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    coordinates: PropTypes.arrayOf(PropTypes.number)
  })).isRequired,
  mapService: PropTypes.shape({
    render: PropTypes.func
  }).isRequired
};
