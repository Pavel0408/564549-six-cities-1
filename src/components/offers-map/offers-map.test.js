import React from 'react';
import renderer from 'react-test-renderer';

import {OffersMap} from "./offers-map";

const offersMock = [
  {
    name: `Beautiful & luxurious apartment at great location`,
    image: `img/apartment-01.jpg`,
    price: 120,
    rating: 5,
    isPremium: false,
    isFavorite: true,
    coordinates: [52.3909553943508, 4.85309666406198]
  },
  {
    name: `Wood and stone place`,
    image: `img/room.jpg`,
    price: 80,
    rating: 10,
    isPremium: true,
    isFavorite: false,
    coordinate: [52.369553943508, 4.85309666406198]
  }
];

const mapMethods = {
  createMap() {},
  addOffersPins() {}
};

describe(`testing OffersMap render`, () => {
  it(`OffersMap is render correctly`, () => {
    const tree = renderer.create(<OffersMap
      offers={offersMock}
      mapMethods={mapMethods}
    />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
