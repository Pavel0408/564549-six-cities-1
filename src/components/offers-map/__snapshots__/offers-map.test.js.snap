// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing OffersMap render OffersMap is render correctly 1`] = `
<div
  data-id="Amsterdam"
  id="map"
  style={
    Object {
      "height": 850,
    }
  }
/>
`;
